### PR TITLE
Export API for stallers and remove access to internal deletages of stallers

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

**Comment by songgao** [comment]
I'm making a few more changes so please hold off for reviewing; Thanks!

**Comment by songgao** [comment]
Should be ready for review now!

**Comment by songgao** [comment]
hmm... this `StallBlockOpAll` API isn't quite useful. I'll have to update again ... sorry guys!

**Comment by songgao** [comment]
Just reverted the last commit. I'll include that in a separate PR which will include staller support in dsl tests as well. So this is open for review again. Really sorry about the comment storm!

**Comment by strib** [comment]
This is beautiful and made me weep with joy.  Nice cleanup!  Looks good to me.

**Comment by songgao** [comment]
Thanks for the kind words @strib !

File: libkbfs/test_stallers.go [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+	id BlockID, plainSize int, readyBlockData ReadyBlockData, err error) {
+	f.maybeStall(ctx, StallableBlockReady)
+	return f.delegate.Ready(ctx, md, block)
+}
+
+func (f *stallingBlockOps) Put(
+	ctx context.Context, md *RootMetadata, blockPtr BlockPointer,
+	readyBlockData ReadyBlockData) error {
+	f.maybeStall(ctx, StallableBlockPut)
+	select {
```
Comment (akalin-keybase) [inline libkbfs/test_stallers.go]: Looks like we do this (check `ctx.Done()`) inconsistently in the methods. Not necessary for this PR, but may be worth doing this for all the methods that normally involve RPCs.

Comment (songgao) [inline libkbfs/test_stallers.go]: Oops sorry I merged too fast. I'll push a separate PR for this.

**Comment by akalin-keybase** [comment]
lgtm, thanks for the cleanup!

File: libkbfs/test_stallers.go (hunk 1 above)
Comment (strib) [inline libkbfs/test_stallers.go]: It's possible there are some tests that rely on context checking and some that don't, for different methods.  We'd have to clean that stuff up as well.  I don't think it's worth doing for this PR.

**Pull request merged**